include/kodlab_mjbots_sdk/common_header.h

```diff
@@ -31,4 +33,84 @@ static void EnableCtrlC() {
   sigaction(SIGINT, &sigIntHandler, NULL);
   CTRL_C_DETECTED = false;
 }
+
+/**
+ * @brief Simple object wrapper that caches the latest value until invalidated
+ * @tparam T typename
+ */
+template<typename T>
+class ValidatedCache
+{
+
+private:
+  /**
+   * @brief Validation status of \c data_
+   */
+  bool valid_;
+
+  /**
+   * @brief Data being stored
+   */
+  T data_;
+
+public:
+  /**
+   * @brief Invalid data constructor
+   */
+  ValidatedCache() : valid_(false) {}
+
+  /**
+   * @brief Valid data constructor
+   */
+  explicit ValidatedCache(const T &data) : data_(data), valid_(true) {}
```

Comment:
Add one more constructor (per comments in attitude class) that allows the setting of a default "zero" or initial value that is valid or not, i.e. explicit ValidatedCache(const T &data, const bool valid) : data_(data), valid_(valid) {}

Reply:
Resolved in ced2bfa83422a5806f829c57fa072a9ec6c14f7b